QuantumExperience/QuasmDriver.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Quantum.Primitive;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Quasm
+{
+    /// <summary>
+    /// Quick and dirty Simulatorbase
+    /// Just enough to show that it would work
+    /// Please don't put this in production until its fully engineerd.
+    /// This code could eat your cat. So imagine what schrodinger has to say about that one.
+    /// </summary>
+    public abstract class QuasmDriver : SimulatorBase
+    {
+        protected abstract List<Result> RunQuasm(StringBuilder quasm);
+        public abstract int QBitCount { get; }
+       
+        public class QSimH : H
+        {
+            public QSimH(IOperationFactory m) : base(m)
+            {
+            }
+
+            public override Func<Qubit, QVoid> Body
+            {
+                get
+                {
+                    return delegate (Qubit q1)
+                    {
+                        if (q1 == null)
+                        {
+                            return QVoid.Instance;
+                        }
+                        QuasmLog.AppendLine("h q[" + (uint)q1.Id + "];");
+                        return QVoid.Instance;
+                    };
+                }
+            }
+        }
+
+        public override void StartOperation(string operationName, OperationFunctor functor, object inputValue)
```

Comment:
This is supposed to be triggered on every operation, not only user defined ones. The fact that calling X or H doesn't trigger it is a bug, which has been fixed internally.
I don't think you want to put this on every call. I would suggest putting it on the constructor of the simulator.

Reply:
Check, will be moving that.